packages/js/product-editor/src/components/empty-state/empty-state.tsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
+
+type EmptyStateProps = React.DetailedHTMLProps<
+	React.HTMLAttributes< HTMLDivElement >,
+	HTMLDivElement
+> & {
+	names: string[];
+};
+
+export function EmptyState( { names = [], ...props }: EmptyStateProps ) {
+	return (
+		<div
+			{ ...props }
+			role="none"
+			className="woocommerce-product-empty-state"
+		>
+			{ names.map( ( name ) => (
+				<div
+					key={ name }
+					className="woocommerce-product-empty-state__row"
+				>
+					{ name === '' ? (
+						<div className="woocommerce-product-empty-state__name" />
+					) : (
+						<div>{ name }</div>
+					) }
```

Comment:
To make this more defensive, when the `name` is an empty string we use a grey block.